Sillycore.EntityFramework/DataContextBase.cs

```diff
@@ -1,26 +1,36 @@
 ﻿using System;
 using System.Collections.Generic;
 using System.Data;
-using System.Linq;
 using System.Reflection;
 using System.Threading;
 using System.Threading.Tasks;
 using Microsoft.EntityFrameworkCore;
-using Microsoft.EntityFrameworkCore.ChangeTracking;
 using Sillycore.Domain.Abstractions;
 using Sillycore.EntityFramework.Attributes;
+using Sillycore.EntityFramework.Extensions;
 using Sillycore.EntityFramework.Mapping;
 
 namespace Sillycore.EntityFramework
 {
-    public abstract class DataContextBase : DbContext
+    public abstract class DataContextBase : DbContext, IEntityEventPublisher
     {
         private long _inMemorySequenceId;
+        internal List<IEntityEventListener> EventListeners { get; }
+        protected internal bool SetUpdatedOnSameAsCreatedOnForNewObjects { get; set; }
 
-        protected bool SetUpdatedOnSameAsCreatedOnForNewObjects { get; set; }
+        protected DataContextBase(DbContextOptions options, SillycoreDataContextOptions sillycoreDataContextOptions)
+            : base(options)
+        {
+            EventListeners = new List<IEntityEventListener>();
+            if (sillycoreDataContextOptions.UseDefaultEventListeners)
+            {
+                EventListeners.Add(new AuditEventListener());
+                EventListeners.Add(new SoftDeleteEventListener());
+            }
+        }
 
         protected DataContextBase(DbContextOptions options)
-            : base(options)
+            : this(options, new SillycoreDataContextOptions())
```

Comment:
Incorrent usage due to misleading property UseDefaultEventListeners.
Since default value is gonna be false this feature are not gonna attempt to include these so called " Default". If it's default then it should be included.

Reply:
in SillycoreDataContextOptions class we set UseDefaultEventListeners to be true initially. but it is misleading a bit.